### PR TITLE
Fix typo in nakamoto tenures table

### PR DESCRIPTION
**Pull request by ASuciuX**
### Description
- fix cause typo
- add checker strict 
- add drop table

### Applicable issues
- fixes https://github.com/stacks-network/stacks-core/issues/4786

### CODE REVIEW

**Comment by ASuciuX** [comment]
Hey @kantai, @obycode, do we need to add an extra if statement for the drop, something like `drop table only if column schema-version doesn't exist`?

Also, how could I test this? I ran `RUST_BACKTRACE=full STACKS_LOG_DEBUG=1 BITCOIND_TEST=1 cargo test -- --test-threads 1` and it failed, but also the `develop` branch from which I've started failed the same way.
```bash
thread 'tests::should_succeed_mining_valid_txs' panicked at testnet/stacks-node/src/node.rs:339:10:
FATAL: failed to initiate mempool: Other("Failed to open chainstate: ClarityError(Interpreter(Interpreter(MarfFailure(\"unable to open database file: /tmp/stacks-node-tests/integrations-neon/68456e5053d7c4e4-1716316920/mocknet/chainstate/vm/clarity/marf.sqlite\"))))")
```

**Review by kantai — COMMENTED** [review]
Thanks for this @ASuciuX

Yes, this schema update needs to be controlled via the schema versioning.

You should:

1. Rename `NAKAMOTO_TENURES_SCHEMA` to `NAKAMOTO_TENURES_SCHEMA_1`
2. Create a new `NAKAMOTO_TENURES_SCHEMA_2` which includes the `DROP TABLE` and the new `CREATE TABLE`.
3. Create a NAKAMOTO_CHAINSTATE_SCHEMA_2 variable in `stackslib/src/chainstate/nakamoto/mod.rs` which includes the `NAKAMOTO_TENURES_SCHEMA_2`
4. Update `CHAINSTATE_VERSION` to `5` in `stackslib/src/chainstate/stacks/db/mod.rs`
5. Add a new `"4"` case to the match in `apply_schema_migrations` in `stackslib/src/chainstate/stacks/db/mod.rs`, which applies `NAKAMOTO_CHAINSTATE_SCHEMA_2`.

**Comment by ASuciuX** [comment]
Thank you very much @kantai! How could I test what I added so far?

I also looked over other tables and I spotted [here](https://github.com/stacks-network/stacks-core/blob/b7f7295775ba8571ed322846367451fc6bf44126/stackslib/src/chainstate/nakamoto/mod.rs#L151-#L171) that on one table (different table than the one this issue is pointing to), some fields are declared as `INT` and others as `INTEGER`? I've looked through the docs and saw they both result in `INTEGER`. If we consider adding the `STRICT` tag to it theoretically it shouldn't result in the affinity as `INT` it's not a 1-on-1 match with `INTEGER`. [[doc link]](https://www.sqlite.org/datatype3.html#determination_of_column_affinity) .

File: stackslib/src/chainstate/stacks/db/mod.rs [hunk 1]
```diff
@@ -1079,6 +1079,13 @@ impl StacksChainState {
                             tx.execute_batch(cmd)?;
                         }
                     }
+                    "4" => {
+                        // migrate to nakamoto 2
+                        info!("Migrating chainstate schema from version 4 to 5: fix nakamoto tenure typo");
```
Comment (ASuciuX) [inline stackslib/src/chainstate/stacks/db/mod.rs:1084]: Do we want to place another message here?

**Comment by ASuciuX** [comment]
According to the [docs](https://www.sqlite.org/stricttables.html#:~:text=SQLite%20supports%20a%20strict%20typing%20mode%2C%20as%20of%20version%203.37.0%20(2021%2D11%2D27)%2C%20that%20is%20enabled%20separately%20for%20each%20table), the first sqlite_version that supports `STRICT` is `3.37.0`. From the `stacks-core`, the sqlite_version is `3.33.0` and the tests were failing with this output:
```
Migrating chainstate schema from version 4 to 5: fix nakamoto tenure typo
thread 'chainstate::nakamoto::tests::test_make_miners_stackerdb_config' panicked at stackslib/src/net/mod.rs:2452:14:
called `Result::unwrap()` on an `Err` value: DBError(SqliteError(SqliteFailure(Error { code: Unknown, extended_code: 1 }, Some("near \"STRICT\": syntax error"))))
```

I would remove from this issue the `STRICT` and add it after we update the sqlite_version.

File: stacks-signer/src/signerdb.rs [hunk 2]
```diff
@@ -144,6 +144,17 @@ impl SignerDb {
         )
     }
 
+     /// Get the sqlite version from the database
+     pub fn get_sqlite_version(
```
Comment (ASuciuX) [inline stacks-signer/src/signerdb.rs:148]: I've added this method to display the sqlite version used. I'm assuming these DBs should have the same version.

**Pull request merged**